### PR TITLE
Fix downstream bug in IL

### PR DESCRIPTION
**Pull request by JimmyYang20**
bug: after deleting dataset, the event of deleting job can be sent to LC.

Signed-off-by: JimmyYang20 <yangjin39@huawei.com>

### CODE REVIEW

File: pkg/globalmanager/controllers/incrementallearning/downstream.go [hunk 1]
```diff
@@ -62,18 +62,15 @@ func (c *Controller) syncToEdge(eventType watch.EventType, obj interface{}) erro
...
-		return nil
-	}
-
 	dataName := job.Spec.Dataset.Name
+	// LC has dataset object on this node that may call dataset node
+	var dsNodeName string
 	ds, err := c.client.Datasets(job.Namespace).Get(context.TODO(), dataName, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("dataset(%s/%s) not found", job.Namespace, dataName)
+		klog.Errorf("not found job(name=%s/%s)'s dataset, error: %v", job.Kind, job.Name, err)
```
Comment (Poorunga) [inline pkg/globalmanager/controllers/incrementallearning/downstream.go:70]: There is no return here, maybe dsNodeName will be empty in the function below:
```
	if eventType == watch.Deleted {
		// delete jobs from all LCs
		nodes := sets.NewString(dsNodeName, trainNodeName, evalNodeName, deployNodeName)
		for node := range nodes {
			c.sendToEdgeFunc(node, eventType, job)
		}
		return nil
	}
```
`dsNodeName` in `sets.NewString(dsNodeName, trainNodeName, evalNodeName, deployNodeName)` maybe an empty string.

Comment (JimmyYang20) [reply pkg/globalmanager/controllers/incrementallearning/downstream.go:70]: the func `sendToEdgeFunc` will handle the case where the node is empty.

**Pull request merged**